### PR TITLE
Simple strangler pattern

### PR DESCRIPTION
**Pull request by ZhouSky**
Based on issue #870 , simple demo and description for strangler pattern.

### CODE REVIEW

File: strangler/src/main/java/com/iluwater/strangler/App.java [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ */
+
+package com.iluwater.strangler;
```
Comment (iluwatar) [inline strangler/src/main/java/com/iluwater/strangler/App.java:24]: Please use package `com.iluwatar`

File: strangler/src/main/java/com/iluwater/strangler/HalfArithmetic.java [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+/**
+ * System under migration. Depends on old version source ({@link OldSource}) and
+ * developing one ({@link HalfSource}).
+ */
+public class HalfArithmetic {
+  private static final Logger LOGGER = LoggerFactory.getLogger(HalfArithmetic.class);
+  private static final String VERSION = "1.5";
+
+  private HalfSource newSource;
+  private OldSource oldSource;
```
Comment (iluwatar) [inline strangler/src/main/java/com/iluwater/strangler/HalfArithmetic.java:38]: I would make these private members final

File: strangler/src/main/java/com/iluwater/strangler/HalfArithmetic.java [hunk 3]
```diff
@@ -0,0 +1,74 @@
...
+
+  public HalfArithmetic(final HalfSource newSource, final OldSource oldSource) {
```
Comment (iluwatar) [inline strangler/src/main/java/com/iluwater/strangler/HalfArithmetic.java:40]: Using final in method parameters has no real significance

Comment (iluwatar) [reply strangler/src/main/java/com/iluwater/strangler/HalfArithmetic.java:40]: Usually you can just omit the final in method parameters. See for example here https://stackoverflow.com/questions/2236599/final-keyword-in-method-parameters

**Comment by iluwatar** [comment]
There's a conflict preventing merge.

**Comment by iluwatar** [comment]
@ZhouSky please comment when you're ready for another review.

File: strangler/src/main/java/com/iluwater/strangler/App.java (hunk 1 above)
Comment (ZhouSky) [reply strangler/src/main/java/com/iluwater/strangler/App.java]: ..It's my fault.. I'm sorry for that.

File: strangler/src/main/java/com/iluwater/strangler/HalfArithmetic.java (hunk 2 above)
Comment (ZhouSky) [reply strangler/src/main/java/com/iluwater/strangler/HalfArithmetic.java]: I personally think the sources should can be changed for strangler pattern, so originally want to make it public. But it just a simple demo, so I follow your advice.

File: strangler/src/main/java/com/iluwater/strangler/HalfArithmetic.java (hunk 3 above)
Comment (ZhouSky) [reply strangler/src/main/java/com/iluwater/strangler/HalfArithmetic.java]: Thanks for your advice, I am indeed not familiar with final, but the final here added according to PMD suggestions.

**Comment by ZhouSky** [comment]
I am sorry that I don't know how to deal with a code review gracefully, so it seems I raise a new pull request #1250. I am really sorry about inconvenience caused by me...

**Comment by iluwatar** [comment]
Ok, I'll close this PR

**Pull request closed without merging**